Handle API change in OpenSSL 0.9.7: an X509V3_EXT_METHOD can have an "it"
member which must be freed using its own method pointer instead of the method
structure's free method.

Based on patch from https://bugzilla.redhat.com/bugzilla/show_bug.cgi?111492,
omitting the reversion of an upstream fix which was included in 2.0.8.

--- openldap-2.0.27/libraries/libldap/tls.c	2002-01-26 14:55:02.000000000 -0500
+++ openldap-2.0.27/libraries/libldap/tls.c	2004-05-18 13:38:44.000000000 -0400
@@ -842,7 +842,12 @@
 		}
 	    }
 	    method = X509V3_EXT_get(ex);
-	    method->ext_free(alt);
+#if (OPENSSL_VERSION_NUMBER >= 0x00907000L)
+	    if (method->it)
+	        ASN1_item_free(alt, ASN1_ITEM_ptr(method->it));
+	    else
+#endif
+	        method->ext_free(alt);
 	    if (i < n)	/* Found a match */
 		ret = LDAP_SUCCESS;
 	}
